tofhir-engine/src/main/scala/io/tofhir/engine/execution/FileStreamInputArchiver.scala

```diff
@@ -181,10 +180,10 @@ class StreamingArchiverTask(archiver: FileStreamInputArchiver, runningJobRegistr
   override def run(): Unit = {
     // Get executions with streaming queries and file system sources and apply
     val executions = runningJobRegistry.getRunningExecutionsWithCompleteMetadata()
-      .filter(execution => execution.isStreaming())
-      .filter(execution => execution.job.sourceSettings.head._2.isInstanceOf[FileSystemSourceSettings])
+      .filter(execution => execution.isStreamingJob)
+      .filter(execution => execution.hasFileSystemSource)
     executions.foreach(execution => {
-      if (execution.isStreaming()) {
+      if (execution.isStreamingJob) {
```

Comment:
I think this filter is redundant, we already filter the streaming job in executions variable.